src/components/Pagination/Pagination.tsx

```diff
@@ -32,145 +32,104 @@ export type PaginationProps = {
+function Pagination(props: PaginationProps) {
+  const handleClick = (pageNumber: number, event: MouseEvent) => {
     if (isSpecialClick(event)) {
       // do not alter the default browser behavior
       // if one special key is down
       return;
     }
     event.preventDefault();
-    this.props.setCurrentPage(pageNumber);
-  };
-
-  private previousPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Previous',
-      additionalClassName: this.props.cssClasses.previousPageItem,
-      isDisabled: this.props.isFirstPage,
-      label: this.props.templates.previous,
-      pageNumber: this.props.currentPage - 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private nextPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Next',
-      additionalClassName: this.props.cssClasses.nextPageItem,
-      isDisabled: this.props.isLastPage,
-      label: this.props.templates.next,
-      pageNumber: this.props.currentPage + 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private firstPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'First',
-      additionalClassName: this.props.cssClasses.firstPageItem,
-      isDisabled: this.props.isFirstPage,
-      label: this.props.templates.first,
-      pageNumber: 0,
-      createURL: this.props.createURL,
-    });
+    props.setCurrentPage(pageNumber);
   };
 
-  private lastPageLink = () => {
-    return this.pageLink({
-      ariaLabel: 'Last',
-      additionalClassName: this.props.cssClasses.lastPageItem,
-      isDisabled: this.props.isLastPage,
-      label: this.props.templates.last,
-      pageNumber: this.props.nbPages! - 1,
-      createURL: this.props.createURL,
-    });
-  };
-
-  private pages = () => {
-    return this.props.pages!.map((pageNumber) =>
-      this.pageLink({
-        ariaLabel: `${pageNumber + 1}`,
-        additionalClassName: this.props.cssClasses.pageItem,
-        isSelected: pageNumber === this.props.currentPage,
-        label: `${pageNumber + 1}`,
-        pageNumber,
-        createURL: this.props.createURL,
-      })
-    );
-  };
-
-  public render() {
-    return (
-      <div
-        className={cx(this.props.cssClasses.root, {
-          [this.props.cssClasses.noRefinementRoot]: this.props.nbPages! <= 1,
-        })}
-      >
-        <ul className={this.props.cssClasses.list}>
-          {this.props.showFirst && this.firstPageLink()}
-          {this.props.showPrevious && this.previousPageLink()}
-          {this.pages()}
-          {this.props.showNext && this.nextPageLink()}
-          {this.props.showLast && this.lastPageLink()}
-        </ul>
-      </div>
-    );
-  }
+  return (
+    <div
+      className={cx(props.cssClasses.root, {
+        [props.cssClasses.noRefinementRoot]: props.nbPages! <= 1,
+      })}
+    >
+      <ul className={props.cssClasses.list}>
+        {props.showFirst && (
+          <PaginationLink
+            key="first"
+            ariaLabel={'First'}
```

Comment:
This is the syntax we use everywhere else. (Same applies in other places.)

```suggestion
            ariaLabel="First"
```